R/generate_template.R

```diff
@@ -313,3 +345,53 @@ create_new_application <- function(name, location, sampleapp = FALSE, resetbutto
 
     return()
 }
+
+create_default_theme_file <- function(theme_file) {
+    lines <- c("### primary_color",
+               "# To set primary status color, that affect the color of header, valueBox, infoBox and box",
+               "# Valid possible value are colors names or hex-decimal value (i.e,: \"blue\", \"#086A87\")",
+               "# Blank/empty value will use default value",
+               "primary_color: \n\n",
+               
+               
+               "# Sidebar variables: change the default sidebar width, colors:",
+               "### sidebar_width",
+               "# width is numeric value in pixels, must be greater than 0",
+               "# Valid possible value are 200, 350, 425, ...",
+               "# Blank/empty value will use default value",
+               "sidebar_width: \n",
+               
+               "### sidebar_background_color",
+               "# Valid possible value are colors names or hex-decimal value (i.e,: \"blue\", \"#086A87\")",
+               "# Blank/empty value will use default value",
+               "sidebar_background_color: \n",
+               
+               "### sidebar_item_background_hover_color",
+               "# Valid possible value are colors names or hex-decimal value (i.e,: \"blue\", \"#086A87\")",
+               "# Blank/empty value will use default value",
+               "sidebar_item_background_hover_color: \n",
+               
+               "### sidebar_color",
+               "# Valid possible value are colors names or hex-decimal value (i.e,: \"blue\", \"#086A87\")",
+               "# Blank/empty value will use default value",
+               "sidebar_color: \n\n",
+               
+               "# body variables",
+               "### body_background_color",
+               "# change the default width of the body as well as its color:",
+               "# Valid possible value are colors names or hex-decimal value (i.e,: \"blue\", \"#086A87\")",
+               "# Blank/empty value will use default value",
+               "body_background_color: \n",
+               
+               "# boxes variables",
+               "### box_background_color",
+               "# Valid possible value are colors names or hex-decimal value (i.e,: \"blue\", \"#086A87\")",
```

Comment:
Since this is repeating for every attribute - Can add this to the top of the file to indicate the valid format of the colors and width. 

For individual fields  - We can only specify the significance /purpose of the field.

Reply:
There is no general documentation for the file and from banner feature in GECO notes from Connie, it is preferred for each item to have its complete documentation.